### DISCUSSION TITLE
choose region that malloc address will work?

### DISCUSSION THREAD

**Discussion by RodrigoDornelles** [cc65/cc65#1883] [Q&A] [answered]
I'm intending to manually manage part of my memory, but I find the malloc provided by cc65 quite nice for other things, I don't want to take all control of the heap, would it be possible for me to delimit an exclusive region for malloc and the other I take control?

I'm using target for NES.

 * PS. Sorry if it's a noob question, I don't know if it's possible, or if it would be a feature request.

**Comment by mrdudz** [comment, marked as answer]
Usually the heap grows downward from the end of the RAM section (see nes.cfg). The easiest way to do this would be making the RAM segment smaller and manage the region at the end of the memory manually. For more complex setups you might have to replace the function that initializes the heap and make it use an arbitrary region.

**Reply by RodrigoDornelles** [reply]
can u tell me where i find the heap initialization function? 👍

**Comment by mrdudz** [comment]
look at crt0.s in libsrc/nes ... it calls initlib, which in turn calls the heap setup function(s)
